external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -395,13 +395,28 @@ private[kafka010] class KafkaOffsetReader(
 
 private[kafka010] object KafkaOffsetReader {
 
-  def kafkaSchema: StructType = StructType(Seq(
-    StructField("key", BinaryType),
-    StructField("value", BinaryType),
-    StructField("topic", StringType),
-    StructField("partition", IntegerType),
-    StructField("offset", LongType),
-    StructField("timestamp", TimestampType),
-    StructField("timestampType", IntegerType)
-  ))
+  def kafkaSchema(includeHeaders: Boolean): StructType = {
+    if (includeHeaders) {
+      StructType(Seq(
+        StructField("key", BinaryType),
+        StructField("value", BinaryType),
+        StructField("topic", StringType),
+        StructField("partition", IntegerType),
+        StructField("offset", LongType),
+        StructField("timestamp", TimestampType),
+        StructField("timestampType", IntegerType),
+        StructField("headers", MapType(StringType, BinaryType))
+      ))
+    } else {
+      StructType(Seq(
+        StructField("key", BinaryType),
```

Comment:
Don't repeat all this. If includeHeaders, add one element to the Seq, then make a structtype from it.